packages/@lwc/module-resolver/src/index.ts

```diff
@@ -61,48 +67,76 @@ function resolveModulesFromDir(modulesDir: string): RegistryEntry[] {
     return resolvedModules;
 }
 
-function resolveModulesFromNpm(packageName: string): RegistryEntry[] {
-    let resolvedModules: RegistryEntry[] = [];
-    try {
-        const pkgJsonPath = require.resolve(`${packageName}/package.json`);
-        const packageDir = path.dirname(pkgJsonPath);
-        const lwcConfigFile = path.join(packageDir, LWC_CONFIG_FILE);
+function resolveModulesFromNpm(
+    npmModuleRecord: NpmModuleRecord,
+    opts: InnerResolverOptions
+): RegistryEntry[] {
+    const { npm, map: aliasMapping } = npmModuleRecord;
+    const pkgJsonPath = require.resolve(`${npm}/package.json`, { paths: [opts.rootDir] });
+    const packageDir = path.dirname(pkgJsonPath);
+    const lwcConfig = getLwcConfig(packageDir);
+    const { version } = loadPackageJson(packageDir);
+    const innerOpts = { ...opts, rootDir: packageDir, scopeDir: opts.rootDir, version };
+    const resolvedModules: RegistryEntry[] = [];
 
-        if (fs.existsSync(lwcConfigFile)) {
-            resolvedModules = resolveModules({ rootDir: lwcConfigFile });
+    if (lwcConfig && lwcConfig.modules) {
+        validateNpmConfig(lwcConfig);
+        const exposedModules = lwcConfig.expose || [];
+        const modules = resolveModulesFromList(lwcConfig.modules, innerOpts);
+        if (!opts.skipExposeFilter) {
+            resolvedModules.push(...modules.filter(m => exposedModules.includes(m.specifier)));
         } else {
-            const pkgJson = JSON.parse(fs.readFileSync(pkgJsonPath, 'utf8'));
-            if (pkgJson.lwc) {
-                resolvedModules = resolveModulesFromList(pkgJson.lwc.modules, { root: packageDir });
-            }
+            resolvedModules.push(...modules);
+        }
+
+        // Remapping some of the modules exposed
+        if (aliasMapping) {
+            resolvedModules.forEach(resolvedModule => {
+                Object.keys(aliasMapping).forEach(oldSpecififer => {
+                    if (resolvedModule.specifier === oldSpecififer) {
+                        resolvedModule.specifier = aliasMapping[oldSpecififer];
+                    }
+                });
+            });
         }
-    } catch (e) {
-        /*noop*/
     }
 
     return resolvedModules;
 }
 
+function resolveModulesFromAlias(
+    moduleRecord: AliasModuleRecord,
+    opts: InnerResolverOptions
+): RegistryEntry | undefined {
+    const { name: specifier, path: modulePath } = moduleRecord;
+    const entry = path.resolve(opts.rootDir, modulePath);
+    if (fs.existsSync(entry)) {
```

Comment:
I don't get this, we need access to the file system, how are we resolving anything on the directories otherwise?

Reply:
I don't think we should conditionally return a module entry depending if the file is present or not, in the case of an AliasModuleRecord. If the config states that a specific module identifier maps to a specific location, a file is expected at this specific location. By adding a condition on the file existence check, the resolver will swallow the error.

We have 2 options here:
- either we throw an error here saying that this file can't be resolved at the expected location
- or we resolve the file location and let the system using the module resolver throws an error if the file is not present at the location.

The `getModuleEntry` suffers from the same issue. We need to make sure that error handling is done the same way across the board.

Reply:
I agree. I encountered a similar swallowed error in `create-lwc-app`: https://github.com/muenzpraeger/create-lwc-app/pull/444 

It took a non-trivial amount of time, and had to step into the `create-lwc-app` code to discover the swallowed error. From a DX perspective, we should tell the user that they have misconfigured the path because the file can't be found.